src/XRP/issued-currency-client.ts

```diff
@@ -783,10 +853,25 @@ export default class IssuedCurrencyClient {
       const rawSendMaxValue = (1 + transferFee / 100) * numericValue
       const numDigitsLeftOfDecimal = String(Math.ceil(rawSendMaxValue)).length
       const allowedDigitsRightOfDecimal = 15 - numDigitsLeftOfDecimal
-      const sendMaxValue = String(
+      const calculatedSendMaxValue = String(
         rawSendMaxValue.toFixed(allowedDigitsRightOfDecimal),
       )
 
+      const sendMaxIssuedCurrencyAmount = new IssuedCurrencyAmount()
+      sendMaxIssuedCurrencyAmount.setCurrency(currencyProto)
+      sendMaxIssuedCurrencyAmount.setIssuer(issuerAccountAddress)
+      sendMaxIssuedCurrencyAmount.setValue(calculatedSendMaxValue)
+
+      const sendMaxCurrencyAmount = new CurrencyAmount()
+      sendMaxCurrencyAmount.setIssuedCurrencyAmount(sendMaxIssuedCurrencyAmount)
+
+      const sendMax = new SendMax()
+      sendMax.setValue(sendMaxCurrencyAmount)
+
+      payment.setSendMax(sendMax)
+    }
+    // manual specification of sendMax value
+    if (sendMaxValue !== undefined) {
```

Comment:
these two logical statements are huge and mutually exclusive; it seems like at the very least there should be an `else` here?

also, the code in these two else blocks is identical. it seems like you could probably do something like

```
// This would go outside of this function
private function calculateSendMax(transferFee: string): string {
  // do the calculation
  return transferFee;
}

// in issuedCurrencyPayment
...
const sendMaxAmount = transferFee !== undefined
  ? this.calculateSendMax(transferFee)
  : sendMaxValue

// construct sendmax
// set it on the payment
```

Reply:
mmm yes, this is a great idea.  and that'll actually allow me to write some unit tests specifically for the sendMax calculation which I think I need to do.